qa/rolling-upgrade/src/test/java/org/opensearch/neuralsearch/bwc/SemanticSearchIT.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.bwc;
+
+import com.carrotsearch.randomizedtesting.RandomizedTest;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import org.opensearch.neuralsearch.TestUtils;
+import static org.opensearch.neuralsearch.TestUtils.NODES_BWC_CLUSTER;
+import static org.opensearch.neuralsearch.TestUtils.TEXT_EMBEDDING_PROCESSOR;
+import org.opensearch.neuralsearch.query.NeuralQueryBuilder;
+
+public class SemanticSearchIT extends AbstractRollingUpgradeTestCase {
+    private static final String PIPELINE_NAME = "nlp-pipeline";
+    private static final String TEST_FIELD = "test-field";
+    private static final String TEXT = "Hello world";
+    private static final String TEXT_MIXED = "Hello world mixed";
+    private static final String TEXT_UPGRADED = "Hello world upgraded";
+    private static final int NUM_DOCS_PER_ROUND = 1;
+
+    // Test rolling-upgrade Semantic Search
+    // Create Text Embedding Processor, Ingestion Pipeline and add document
+    // Validate process , pipeline and document count in rolling-upgrade scenario
+    public void testSemanticSearch() throws Exception {
```

Comment:
```suggestion
    @SneakyThrows
    public void testSemanticSearch_E2ETest() {
```

Reply:
testSemanticSearch_E2Etest is little weird name as it has test on both sides start and end.

Reply:
I can add E2E if you want but testSemanticSearch looks more adequate to it

Reply:
actually testSemantic Search doesn't provide any information. Semantic Search is pretty broad term. What you are actually validating is textEmbedding Processor. So better name here could be

`testTextEmbeddingProcessor`

Reply:
Done